letsencrypt/client/plugins/nginx/configurator.py

```diff
@@ -0,0 +1,554 @@
+"""Nginx Configuration"""
+import logging
+import os
+import re
+import shutil
+import socket
+import subprocess
+import sys
+
+import zope.interface
+
+from letsencrypt.acme import challenges
+
+from letsencrypt.client import achallenges
+from letsencrypt.client import constants
+from letsencrypt.client import errors
+from letsencrypt.client import interfaces
+from letsencrypt.client import le_util
+from letsencrypt.client import reverter
+
+from letsencrypt.client.plugins.nginx import dvsni
+from letsencrypt.client.plugins.nginx import parser
+
+
+class NginxConfigurator(object):
+    # pylint: disable=too-many-instance-attributes,too-many-public-methods
+    """Nginx configurator.
+
+    .. todo:: Add proper support for comments in the config
```

Comment:
nginxparser.py ignores comments in the config file, so a file that has comments will not have them anymore after it's been modified by the configurator.

I think this can be easily fixed by adding a rule for "python-style" comments to https://github.com/letsencrypt/lets-encrypt-preview/pull/351/files#diff-cafec352d6b60f07cb9d6f13f1c99d42R28 but have not looked into pyparsing enough to figure out how.